WooCommerce/Classes/ViewRelated/Orders/Fulfillment/FulfillViewController.swift

```diff
@@ -564,6 +568,18 @@ private extension FulfillViewController {
 
         return orderTracking[orderIndex]
     }
+
+    func lookUpProduct(by productID: Int) -> Product? {
```

Comment:
This logic is repeated here and in OrderDetailsViewController. That might signal that it should be encapsulated somewhere else (I know that is a very broad generalisation, but please bear with me).

I would explore moving this logic down to the view model. 

I left another comment related to this below, but the idea is that both the OrderDetailsViewController and the FulfillViewController are doing some logic that might not belong in the UI layer: fetching shipment trackings, status, 
and products. It is safe to expect that there evil be even more.

Syncing that info is something that we could move down to the view model. The view model could expose whatever data needs to be exposed to the view controller, and could also expose closures to execute when the UI needs to be refreshed or an error needs to be presented. (I am not sure if this is something I've actually mentioned already, or has only lived in my mind, but this is a pattern we repeat in almost every view controller: we fetch data, and reload on success or present a notice on error)

If (and I haven't actually tested it, so it might not even be possible) we moved the knowledge about products and how to fetch them to the view model, then we could eventually share that view model between OrderDetailsViewController, FulfillViewController and ProductListViewController, because as far as I understand, all of them share the same data.

With that said, the biggest advantage of extracting this logic from the view controller would be, to me, that we would add a layer of indirection between the UI and the data we render in the UI. That allows us to split whatever logic we went or need to into its own entity (class, cluster, whatnot) and make room for solving cross-cutting concerns and sync issues (like showing the product images and coordinating different stores)

Sorry for the wall of text, I hope it makes some sense...

Reply:
I was thinking the same thing. I'd like to propose: 
1. I merge this PR in, so the product images can ship on time with the rest of the feature release
2. I create a separate PR that makes the OrderDetailsViewModel take on more responsibility. Too much of it currently lives in the VC. Especially after seeing how well it works to have the results controllers in the View Model.